src/Symfony/Component/Messenger/Tests/Asynchronous/Routing/SenderLocatorTest.php

```diff
@@ -36,6 +37,38 @@ public function testItReturnsTheSenderBasedOnTheMessageClass()
         $this->assertEquals(array(), $locator->getSendersForMessage(new SecondMessage()));
     }
 
+    public function testItReturnsTheSenderBasedOnTheMessageParentClass()
+    {
+        $sender = $this->getMockBuilder(SenderInterface::class)->getMock();
+        $container = new Container();
+        $container->set('my_amqp_sender', $sender);
+
+        $locator = new SenderLocator($container, array(
+            DummyMessage::class => array(
+                'my_amqp_sender',
+            ),
+        ));
+
+        $this->assertEquals(array($sender), $locator->getSendersForMessage(new ChildDummyMessage('Hello', array('name' => 'World'))));
```

Comment:
`assertSame` for all the occurrences of `assertEquals`

Reply:
Fixed.